File: app/src/main/java/com/pnuema/bible/ui/viewholders/VerseViewHolder.java
package com.pnuema.bible.ui.viewholders;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import com.pnuema.bible.R;

public class VerseViewHolder extends RecyclerView.ViewHolder {
    private TextView verseText;

    public VerseViewHolder(@NonNull final ViewGroup parent) {
        super(LayoutInflater.from(parent.getContext()).inflate(getLayout(), parent, false));

        verseText = itemView.findViewById(R.id.verseText);
    }

    public void setVerseText(final CharSequence text) {
        verseText.setText(text);
    }

    public static int getType() {
        return getLayout();
    }

    private static int getLayout() {
        return R.layout.listitem_verse;
    }
}

File: app/src/main/kotlin/com/pnuema/bible/android/ui/viewholders/VerseViewHolder.kt
package com.pnuema.bible.android.ui.viewholders

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.pnuema.bible.android.R
import com.pnuema.bible.android.databinding.ListitemVerseBinding
import com.pnuema.bible.android.statics.fromHtml
import com.pnuema.bible.android.ui.utils.VerseUtils.formatted
import com.pnuema.bible.android.ui.viewstates.VerseViewState

class VerseViewHolder(
        parent: ViewGroup,
        private val binding: ListitemVerseBinding = ListitemVerseBinding.inflate(LayoutInflater.from(parent.context), parent, false)
) : RecyclerView.ViewHolder(binding.root) {
    companion object {
        val type: Int
            get() = layout

        private val layout: Int
            get() = R.layout.listitem_verse
    }

    fun bind(verse: VerseViewState) {
        binding.verseText.text = verse.formatted().fromHtml()
    }
}
